extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -258,62 +234,50 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 
 // RestoreRKE2K3SSnapshot is a helper function to restore a snapshot on an RKE2 or k3s cluster. Returns error if any.
 func RestoreRKE2K3SSnapshot(client *rancher.Client, snapshotRestore *rkev1.ETCDSnapshotRestore, clusterName string) error {
-	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
+	_, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
 	if err != nil {
 		return err
 	}
+	steveWithUpdates := existingSteveAPIObject
 
-	clusterObject.Spec.RKEConfig.ETCDSnapshotRestore = snapshotRestore
-
-	logrus.Infof("Restoring snapshot: %v", snapshotRestore.Name)
-	updatedCluster, err := client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, clusterObject)
+	clusterSpec := &provv1.ClusterSpec{}
+	err = steveV1.ConvertToK8sType(steveWithUpdates.Spec, clusterSpec)
 	if err != nil {
 		return err
 	}
 
-	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.OneMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
-		clusterResp, err := client.Steve.SteveType(ProvisioningSteveResouceType).ByID(updatedCluster.ID)
-		if err != nil {
-			return false, err
-		}
-
-		clusterStatus := &apisV1.ClusterStatus{}
-		err = rancherv1.ConvertToK8sType(clusterResp.Status, clusterStatus)
-		if err != nil {
-			return false, err
-		}
+	clusterSpec.RKEConfig.ETCDSnapshotRestore = snapshotRestore
 
-		if clusterResp.ObjectMeta.State.Name != active {
-			return true, nil
-		}
+	steveWithUpdates.Spec = clusterSpec
 
-		return false, nil
-	})
+	logrus.Infof("Restoring snapshot: %v", snapshotRestore.Name)
+	updatedCluster, err := client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, steveWithUpdates)
 	if err != nil {
 		return err
 	}
 
-	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.FifteenMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
+	err = kwait.PollUntilContextTimeout(context.TODO(), 1*time.Second, defaults.TwoMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
For clarification, why was this cut down to two minutes? Context behind the 15 minutes is that at times, RKE2 can take quite a bit of time to fully restore, so that was used as a safeguard to not have false negative tests.

Reply:
this was split into 2 waits; since this part is only waiting on the snapshots to _start_, I cut it down to 2 mins

later (outside of this function) as part of cluster upgrade / cluster validation, the state is checked again where the timeout is longer ~15 mins

I postulate this is more in line with the structure of what the create functions do; the restore has started, which is what we want. Later, we verify that the process(es) are complete in some other validation i.e. in this case its a cluster upgrade check that is done.